Avoid conflict with system ptrace. We don't have PT_DENY_ATTACH

Index: process.c
--- process.c.orig
+++ process.c
@@ -66,16 +66,6 @@ static void prctl(__attribute__((unused)) int x,
 		  __attribute__((unused)) unsigned long y) {}
 #endif
 
-#ifndef USE_PTRACE
-#undef PT_DENY_ATTACH
-#define PT_DENY_ATTACH 0
-static void ptrace(__attribute__((unused)) int x,
-		   __attribute__((unused)) int y,
-		   __attribute__((unused)) int z,
-		   __attribute__((unused)) int w) {}
-#endif
-
-
 #if defined(__linux__) || defined(__CYGWIN__) || (defined(__NetBSD__) && !defined(KERN_PROC_PATHNAME))
 static int pid_to_cmd(pid_t pid, char *cmd, size_t cmd_size)
 {
@@ -161,7 +151,6 @@ bool process_is_same_executable(pid_t pid)
 void process_disable_ptrace(void)
 {
 	prctl(PR_SET_DUMPABLE, 0);
-	ptrace(PT_DENY_ATTACH, 0, 0, 0);
 
 	struct rlimit limit = { 0, 0 };
 	setrlimit(RLIMIT_CORE, &limit);
